enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,262 @@
+release controllers. A new heterogeneous release controller is bootstrapped and displays "4-stable" entries for ART constructed release payloads.
+
+`oc adm release new` would be [enhanced][WRKLDS-370] to allow the creation of manifestlist release payloads. The manifestlist flow would be triggered whenever the CVO image in an imagestream is a manifestlist. If the CVO image is a standard manifest, the generated release payload will also be a manifest. If the CVO image is a manifestlist, the generated release
+payload would be a manifestlist(containing a manifest for each architecture possessed by the CVO manifestlist). In either case, oc adm release new would permit non-CVO component images to be manifest or manifestlists and pass them through directly to the resultant release manifest(s). If a manifestlist release payload is generated, each architecture specific
+release payload manifest will reference the same pullspecs provided in the input imagestream.
+
+Since the internal Openshift registry does not support manifestlists, the release controller would use quay.io. There will be a new architecture option added to Cincinnati called "multi".
+
+### Upgrades - Cincinnati and CVO
+
+In addition to the homogeneous payload graph that we have today, [support][OCPPLAN-7642] needs to be added for an additional graph for heterogeneous payloads which CVO will need to look at to make upgrade recommendations. There are a few upgrade scenarios to consider with the addition of heterogeneous payloads:
+
+- heterogeneous -> heterogeneous payload upgrade
+- homogeneous -> heterogeneous payload upgrade
+- heterogeneous -> homogeneous payload upgrade
+
+The support for homogeneous payloads will exist so customers running disconnected homogeneous clusters do not incur the cost of mirroring the additional architectures if they were to use heterogeneous payloads. This means that in order for CVO to find the current release in the graph, it will have to know about the existing homogeneous graph as well in addition
+to the new heterogeneous graph. There are few options on the best way to make update recommendations especially in the case of upgrading from a homogeneous to a heterogeneous payload:
+
+- Have the heterogeneous graph include the homogeneous releases so CVO can make recommendations based on the current homogeneous release image. This is complicated by the fact that there will be multiple releases with the same version but different payloads. The trigger to update to a heterogeneous payload would be an option provided by the administrator through
+either the ClusterVersion or Infrastructure API to "move" to the heterogeneous graph.
+- Have both the graphs be completely independent of each other and have an option in one of the aforementioned APIs to make the cluster heterogeneous and CVO would request the heterogeneous graph, find the current release version and update to it.
```

Comment:
I was considering a mix of the two:

`oc adm release upgrade --multi-arch -to=<target>`
If `--multi-arch` is specified (or the cluster is already multi-arch), the Cincinnati query will be limited to the `multi` arch graph. If the _current_ cluster version does not exist in the `multi` arch graph with a connection to the target, then the upgrade is not permitted. If the current cluster is homogeneous, the customer will need to continue on in the homogeneous graph until their installed version is represented in the `multi` arch graph.
If the current and target version exist in the multi arch graph, and an upgrade occurs, then the clusterversion is updated with a net-new field to indicate that the cluster is now multi-arch. All subsequent queries will be for multi arch graphs (and ignore whether --multi-arch is specified).

As suggested in a previous comment, this would be a one-way street with no return to homogeneous outside of working with the support organization and forced upgrades.